constraints.txt

```diff
@@ -0,0 +1,168 @@
+#
+# This file is autogenerated by pip-compile with python 3.9
+# To update, run:
+#
+#    pip-compile --extra=test --extra=yamllint --output-file=constraints.txt docs/requirements.txt setup.cfg
```

Comment:
Mixing the docs pins up with the test ones doesn't seem right. Those envs are unrelated and must be kept strictly separate.